pkg/crc/telemetry/telemetry.go

```diff
@@ -0,0 +1,43 @@
+package telemetry
+
+import (
+	"context"
+	"sync"
+)
+
+type contextKey struct{}
+
+var key = contextKey{}
+
+type Properties struct {
+	lock    sync.Mutex
+	storage map[string]interface{}
+}
+
+func (p *Properties) Set(name string, value interface{}) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	p.storage[name] = value
+}
+
+func (p *Properties) Map() map[string]interface{} {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	ret := make(map[string]interface{})
+	for k, v := range p.storage {
+		ret[k] = v
+	}
+	return ret
+}
+
+func ContextWithEmptyProperties(ctx context.Context) context.Context {
+	return context.WithValue(ctx, key, &Properties{storage: make(map[string]interface{})})
+}
+
+func PropertiesFromContext(ctx context.Context) *Properties {
```

Comment:
Not sure the intermediate `Properties` brings much over `telemetry.SetContextProperty(context, "key", value)` and `telemetry.GetContextProperties(context)`. This would be more concise than the current
```
			properties := telemetry.PropertiesFromContext(cmd.Context())
			properties.Set("key", args[0])
```
and all these calls to `PropertiesFromContext` are missing a `nil` check.

Reply:
done :+1: